.github/workflows/ci_tests.yaml

```diff
@@ -54,40 +56,36 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.10', '3.12']
+        python-version: ['3.10', '3.11', '3.12']
         os: [ubuntu-latest, macos-latest, windows-latest]
         # Is it a draft Pull Request (true or false)?
         isDraft:
           - ${{ github.event.pull_request.draft }}
-        # Only run two jobs (Ubuntu + Python 3.10/3.12) for draft PRs
+        # Only run three jobs on Ubuntu for draft PRs
         exclude:
           - os: macos-latest
             isDraft: true
           - os: windows-latest
             isDraft: true
-        # Pair Python 3.10 with the minimum supported versions of NumPy, pandas, Xarray
-        # and Python 3.12 with the latest versions of NumPy, pandas, Xarray
-        # Only install optional packages on Python 3.12
         include:
+          # Python 3.10 + core packages (minimum supported versions) + optional packages (minimum supported versions if any)
           - python-version: '3.10'
             numpy-version: '1.24'
             pandas-version: '=2.0'
             xarray-version: '=2023.04'
-            optional-packages: ''
+            optional-packages: ' contextily geopandas<1 ipython pyarrow rioxarray sphinx-gallery'
+          # Python 3.12 + core packages (latest versions) + optional packages
           - python-version: '3.12'
             numpy-version: '2.1'
             pandas-version: ''
             xarray-version: ''
-            optional-packages: ' contextily geopandas ipython pyarrow rioxarray sphinx-gallery'
-          # The job below is for testing GeoPandas v0.x on Ubuntu.
-          # The python-version here can't be the versions in the matrix.python-version
-          # defined above. Otherwise, other jobs will be overridden by this one.
-          - os: 'ubuntu-latest'
-            python-version: '3.11'  # Can't be 3.10 or 3.12.
-            numpy-version: '1.24'
-            pandas-version: '=2.1'
+            optional-packages: ' contextily geopandas>=1.0 ipython pyarrow rioxarray sphinx-gallery'
+          # Python 3.11 + core packages
+          - python-version: '3.11'
```

Comment:
I'd still prefer to keep the middle Python version (Python 3.11 currently) on Ubuntu only, no need to test on macOS/Windows.